src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1113,6 +1112,54 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
+        /// <exception cref="SecurityTokenException">if the token is a JWE and the  <see cref="JwtHeaderParameterNames.Enc"/> header claim is null or empty.</exception>
+        /// <exception cref="SecurityTokenDecompressionFailedException">if <paramref name="token"/> is a JWE and decompression fails.</exception>
+        /// <exception cref="SecurityTokenDecryptionFailedException">if <paramref name="token"/> is a JWE and was not able to be decrypted.</exception>
+        /// <exception cref="SecurityTokenSignatureKeyNotFoundException"> if the signing key could not be matched or no signing keys were provided.</exception>
+        /// <exception cref="SecurityTokenInvalidSignatureException"> if the signature of <paramref name="token"/> is not properly formatted.</exception>
+        /// <exception cref="SecurityTokenInvalidSignatureException"> if the signature of <paramref name="token"/> cannot be validated with the provided keys.</exception>
+        /// <exception cref="SecurityTokenInvalidSignatureException"> if <paramref name="token"/> is unsigned but <see cref="TokenValidationParameters.RequireSignedTokens"/> is true.</exception>
+        /// <exception cref="SecurityTokenInvalidSignatureException"> if a <see cref="SignatureValidator"/> or <see cref="TokenReader"/> delegate is provided but it fails or doesn't return a <see cref="JsonWebToken"/>.</exception>
+        /// <returns> A <see cref="ClaimsPrincipal"/> that was created from the JWT. Does not include claims found in the JWT header.</returns>
+        /// <remarks> 
+        /// This method calls the <see cref="Validators.ValidateLifetime"/>, <see cref="Validators.ValidateAudience"/>, <see cref="Validators.ValidateIssuer"/>, <see cref="Validators.ValidateTokenReplay(DateTime?, string, TokenValidationParameters)"/>
+        /// and <see cref="Validators.ValidateIssuerSecurityKey(SecurityKey, SecurityToken, TokenValidationParameters)"/> methods. Any of the exceptions thrown by these methods may also be thrown; see <see cref="Validators"/> to examine
+        /// the full call graph.
+        /// </remarks>
+        public ClaimsPrincipal ValidateToken(string token, TokenValidationParameters validationParameters, out SecurityToken validatedToken)
+        {
+            if (string.IsNullOrEmpty(token))
+                throw LogHelper.LogArgumentNullException(nameof(token));
+
+            if (validationParameters == null)
+                throw LogHelper.LogArgumentNullException(nameof(validationParameters));
+
+            if (token.Length > MaximumTokenSizeInBytes)
+                throw LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(TokenLogMessages.IDX10209, token.Length, MaximumTokenSizeInBytes)));
+
+            var tokenValidationResult = ValidateTokenPrivate(token, validationParameters);
+            if (!tokenValidationResult.IsValid)
```

Comment:
We will benefit from introducing a standard method for detecting if the TokenValidationResult is valid.
This will abstract away any new validation conditions that arise.
In the future, we could allow users to hook into this. 

```
internal void ThrowOnTokenValidationError(TokenValidationResult result, string token, CallContext context)
{
	if (result.Exception != null)
    {
    	if (result.Exception is <SomeIdentityModelExceptio>Exception)
        	throw LogHelper.LogExceptionMessage(result.Exception);
        else
            throw LogHelper.LogExceptionMessage(new <SomeIdentityModelException>, LogHelper.FormatInvariant(LogMessages.IDX?????, this, token), result.Exception), context);
    }

    if (!result.IsValid)
    	throw LogHelper.LogExceptionMessage(new <SomeIdentityModelException>(LogHelper.FormatInvariant(LogMessages.IDX???, this, token)), context);
}
```